src/components/KeyboardAwareScrollView/KeyboardAwareFlatList.js

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {FlatList} from 'react-native';
+import KeyboardAwareBase from './KeyboardAwareBase';
+
+/**
+ * @description: A wrapper component which handles the FlatList insets properly when the keyboard is shown and hides the content, scrolling content above the keybaord.
+ * @example: https://github.com/wix/react-native-ui-lib/blob/master/demo/src/screens/componentScreens/KeyboardAwareScrollViewScreen.js
+ */
+export default class KeyboardAwareFlatList extends KeyboardAwareBase {
+  static displayName = 'KeyboardAwareFlatList';
+
+  static PropTypes = {
+    getTextInputRefs: PropTypes.func,
+    onScroll: PropTypes.func
+  };
+
+  static defaultProps = {
+    ...KeyboardAwareBase.defaultProps,
+    getTextInputRefs: () => {
+      return [];
+    }
+  };
+
+  render() {
+    return (
+      <FlatList
+        {...this.props}
+        {...this.style}
+        contentInset={{bottom: this.state.keyboardHeight}}
+        ref={(r) => {
+          this._keyboardAwareView = r;
+        }}
+        onLayout={(layoutEvent) => {
+          this._onKeyboardAwareViewLayout(layoutEvent.nativeEvent.layout);
+        }}
+        onScroll={(event) => {
```

Comment:
Same for the onScroll callback. 
maybe you can unify the logic in KeyboardAwareBase. 
Anyway, avoid passing inline callbacks.

Reply:
Done

Reply:
I see you didn't update `KeyboardAwareListView` with the same fixes.. 
Is that because we're deprecating the component ?

Reply:
You are right, it should be aligned even if it's should not be used, fixed